src/Unused/Util.hs

```diff
@@ -0,0 +1,12 @@
+module Unused.Util
+    ( groupBy
+    ) where
+
+import Data.List (nub)
+
+groupBy :: Eq b => (a -> b) -> [a] -> [(b, [a])]
+groupBy f l =
```

Comment:
Oh, I see the difference is you want the "key" in the output. I was surprised there was no version like that available, best I could find was this http://stackoverflow.com/questions/15412027/haskell-equivalent-to-scalas-groupby

Might be more efficient than yours, but I'm not convinced it's more readable.